### DISCUSSION TITLE
Choose 'starting-frame' for compute(Frame/Joint)Jacobian

### DISCUSSION THREAD

**Discussion by lukas-ramlab** [stack-of-tasks/pinocchio#2336] [Q&A] [answered]
I'm using an installation where the urdf model is not a single tree, but rather branching out. 
Specifically, the work piece is mounted on a robotic table, and the robot is placed on a linear axis.
Hence, from the static base-frame/world the model is branching out.

I would like to compute the Jacobian (and transform) from the robot-to-work frame.

For the transform it is straight forward, as I can compute world-to-robot and world-to-work separately, and then combine the two.

Is there a similar possibility for the Jacobian? An argument that I can give to a function or setting the starting frame?

Thank you very much for any insight.

**Comment by jcarpent** [comment]
Thanks @lukas-ramlab for using Pinocchio.
If I understand well, you would like to compute the kinematic Jacobian between two joints of the kinematic tree?

**Reply by lukas-ramlab** [reply]
Yes exactly. That would be the goal. Is that easily possible through the interface?

**Reply by jcarpent** [reply, marked as answer]
You can do it indirectly, as follows:
```python
pin.computeJointJacobians(model,data,q)
_1M2_ = data.oMi[joint1_id].actInv(data.oMi[joint2_id])
J1 = pin.getJointJacobian(model,data,joint1_id,pin.LOCAL)
J2 = pin.getJointJacobian(model,data,joint2_id,pin.LOCAL)
J = J1 - _1M2_.toActionMatrix() @ J2
``` 
This snippet should be self-understandable.

**Reply by lukas-ramlab** [reply]
Yes, that's what I was looking for! Thank you very much.
